ARCHITECTURE.md

```diff
@@ -0,0 +1,74 @@
+# Exonum Architecture
+
+This document describes the high-level architecture of Exonum. Its target
+audience is the developers of Exonum core. To learn how to *use* Exonum, see
+https://exonum.com/doc/.
+
+This document is provided on the best effort bases -- it is not guaranteed to be
+up-to-date and complete. The purpose of this document is to highlight the most
+important types, modules and functions in the code, to allow new contributors to
+dive into Exonum quickly. It is assumed that the reader is familiar with core
+concepts of Exonum, such as services and transactions.
+
+
```

Comment:
I usually separate sections (`##`) in markdown docs with two blank lines: there's no difference for rendering, but this looks better in raw text form, because it stands out more than just a single blank line between paragraphs.